src/pages/Home.tsx

```diff
@@ -49,6 +57,24 @@ function Home() {
     '0x90F79bf6EB2c4f870365E785982E1f101E93b906',
   ];
 
+  useEffect(() => {
+    updateStatus();
+    const interval = setInterval(() => {
+      updateStatus();
+    }, 5000);
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
+
+  const truncateRegex = /^(0x[a-zA-Z0-9]{3})[a-zA-Z0-9]+([a-zA-Z0-9]{3})$/;
+  const truncateEthAddress = (address: string) => {
+    const match = address.match(truncateRegex);
+    if (!match) return address;
+    return `${match[1]}…${match[2]}`;
+  };
```

Comment:
Recommend moving this to a utils file (doesn't need to be done in this PR).

Reply:
Sorry, that is totally me being lazy ... moved